### PR TITLE
google-java-format 1.9

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by bayandin** [comment]
```
==> /usr/local/Cellar/google-java-format/1.9/bin/google-java-format foo.java
/usr/local/Cellar/google-java-format/1.9/bin/google-java-format: line 3: /usr/local/opt/openjdk/libexec/openjdk.jdk/Contents/Home/bin/java: No such file or directory
/usr/local/Cellar/google-java-format/1.9/bin/google-java-format: line 3: exec: /usr/local/opt/openjdk/libexec/openjdk.jdk/Contents/Home/bin/java: cannot execute: No such file or directory
```

`depends_on "openjdk"` line is missing in the formula

File: Formula/google-java-format.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@ class GoogleJavaFormat < Formula
   end
 
   depends_on "maven" => :build
+  depends_on "openjdk" => :test
```
Comment (bayandin) [inline Formula/google-java-format.rb:16]: I believe we need this in runtime as well

Comment (chenrui333) [reply Formula/google-java-format.rb:16]: I dont like adding java 14 as runtime. If I really need to have openjdk as runtime, I would rather have `openjdk@11`.

**Comment by chenrui333** [comment]
relates to https://github.com/google/google-java-format/issues/508

**Review by SMillerDev — DISMISSED** [review]

File: Formula/google-java-format.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/google-java-format.rb]: Homebrew is an evergreen package manager. We use the latest openjdk if it works

**Comment by bayandin** [comment]
`maven` depends on `openjdk`, so it won't work with `opejdk@11` probably.

Anyway, I would suggest switching formulae to `:bottle_unneded` with jar-file provided by upstream https://github.com/google/google-java-format/releases/tag/google-java-format-1.9 (as mentioned in https://github.com/google/google-java-format/issues/508#issuecomment-663980739 it should work even in java 8)

And we can safely use `depends_on "openjdk"` with `overridable_java_home_env`

**Comment by chenrui333** [comment]
> `maven` depends on `openjdk`, so it won't work with `opejdk@11` probably.
> 
> Anyway, I would suggest switching formulae to `bottle :unneeded` with a jar-file provided by upstream https://github.com/google/google-java-format/releases/tag/google-java-format-1.9 (as mentioned in [google/google-java-format#508 (comment)](https://github.com/google/google-java-format/issues/508#issuecomment-663980739) it should work even in java 8)
> 
> And we can safely use `depends_on "openjdk"` with `overridable_java_home_env`

agree, we should not rely on java 11 for the runtime.

**Comment by chenrui333** [comment]
I will take it back, the upstream jar is compiled with java 11. So we have to use jdk11, anyway, this is already a good achievement of not using java 14 yet.

**Comment by chenrui333** [comment]
error log ref:
```
==> /usr/local/Cellar/google-java-format/1.9/bin/google-java-format foo.java
Picked up _JAVA_OPTIONS: -Duser.home=/Users/rchen/Library/Caches/Homebrew/java_cache -Djava.io.tmpdir=/private/tmp
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: com/google/googlejavaformat/java/Main has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

**Comment by bayandin** [comment]
> error log ref:
> 
> ```
> ==> /usr/local/Cellar/google-java-format/1.9/bin/google-java-format foo.java
> Picked up _JAVA_OPTIONS: -Duser.home=/Users/rchen/Library/Caches/Homebrew/java_cache -Djava.io.tmpdir=/private/tmp
> Error: A JNI error has occurred, please check your installation and try again
> Exception in thread "main" java.lang.UnsupportedClassVersionError: com/google/googlejavaformat/java/Main has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
> ```

This is strange; it says that the jar was compiled with java 11 (class file version 55.0), but it is trying to be run with java 8 (class file version 52.0). Did we get this in our CI?

> I will take it back, the upstream jar is compiled with java 11. So we have to use jdk11

I would not agree with it. 
We don't force users to use any particular version of java, but we ship software with the latest working java version. Users are free to set `JAVA_HOME` to whatever they want and formula will respect this. 
It seems `google-java-format` works with java 14 (at least `brew test google-java-format` worked fine for me locally with java 14).
I tried with this:

```ruby
  depends_on "openjdk"

  def install
    libexec.install "google-java-format-#{version}-all-deps.jar" => "google-java-format.jar"
    bin.write_jar_script libexec / "google-java-format.jar", "google-java-format"
  end
```

**Comment by SMillerDev** [comment]
I think my snippet from the OpenJDK migration times is relevant here. We don't use LTS versions in homebrew, unless that is the latest supported version.

---

Homebrew will use brewed dependencies where possible. The fact that other Java versions were previously possible was an unfortunate side-effect of java not being built from source as a formula. The current situation is the intended usage within Homebrew. If people want to use a different dependency the best course of action is to maintain it in a tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap.

Just a couple of points to highlight:
- The `openjdk` installed as a dependency is not symlinked anywhere and will be used only if `JAVA_HOME` is not set

- If you'd like to use another Java installation feel free to set `JAVA_HOME` to any Java you want

- Using some random/installed by default Java is not reliable for software shipping.

**Comment by chenrui333** [comment]
ok, I have updated the formula to use openjdk, I will see if there is any issue with using this in my CI after the merge.

Thanks @bayandin!

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**